pkg/apis/core/v1beta1/types_controllerregistration.go

```diff
@@ -73,6 +73,9 @@ type ControllerResource struct {
 	// combination. This field is immutable.
 	// +optional
 	Primary *bool `json:"primary,omitempty" protobuf:"varint,5,opt,name=primary"`
+	// Lifecycle defines a strategy that determines when different operations on a ControllerResource should be performed.
+	// +optional
+	Lifecycle *ControllerResourceLifecycle `json:"lifecycle,omitempty" protobuf:"bytes,6,opt,name=lifecycle"`
```

Comment:
I am missing the API defaulting for this field in case the `Kind` is `Extension` (same for `v1alpha1`)

Reply:
What will be the benefits to explicitly default this? Right now the default behavior is determined by the `pkg/operation/botanist/component/extensions/extension/extension.go` package which handles the case when these values are not set.

Reply:
The benefit is that it is clearer for the user to get to know what is actually going on "by default". Such internal behaviour must be looked up in code or documentation.

Reply:
As far as I can tell right now we also do not explicitly default `GloballyEnabled` and `ReconcileTimeout` but they still have some internal behavior and defaulting. IMO as a user I can create the `ControllerRegistration` and still not know what was defaulted and what was not. For me it does make sense to not explicitly set this since the defaults will cover probably 90%+ of the cases and the user should not really care in most scenarios. If the user cares and needs different behavior then one can read the documentation and enable a specific flow.

If you really insist on this comment I will make the change, although I am not completely sure that this is the better option. Then it will also probably make sense to default the other two configurations too. WDYT?

Reply:
I don't think the other two fields were *intentionally* not defaulted. At some point in time we agreed to follow this approach for newly introduced fields, but we didn't do it for all already existing optional fields. From https://github.com/gardener/gardener/blob/master/docs/development/changing-the-api.md#gardener-api:

> 3. If necessary then implement/adapt defaulting logic defined in the versioned APIs (e.g., pkg/apis/core/v1beta1/defaults*.go).

While "if necessary" can be interpreted quite freely, it actually means "if the field is optional".
So yes, if you touch it now anyways then it makes sense to also include `globallyEnabled` and `reconcileTimeout` 👍🏻

Reply:
Ok, I understand it more clearly now. If we have some agreement that this convention should be followed then I have no problem sticking to it. I will adapt with the suggestions and also add defaulting for `globallyEnabled` and `reconcileTimeout`.